leftwm-core/src/handlers/command_handler.rs

```diff
@@ -143,6 +143,38 @@ fn toggle_state(state: &mut State, window_state: WindowState) -> Option<bool> {
     let window = state.focus_manager.window(&state.windows)?;
     let handle = window.handle;
     let toggle_to = !window.has_state(&window_state);
+    let tag_id = state.focus_manager.tag(0)?;
+
+    let for_active_workspace = |x: &Window| -> bool { x.tag == Some(tag_id) && x.is_managed() };
+
+    if window_state == WindowState::Fullscreen {
+        //Going fullscreen, so we save the window order
+        //or else, we restore it!
+        if toggle_to {
+            let mut to_reorder = helpers::vec_extract(&mut state.windows, for_active_workspace);
+            let handles = to_reorder
+                .iter()
+                .map(|each_handle| each_handle.handle)
+                .collect::<Vec<_>>();
+
+            state.window_order = handles;
+            state.windows.append(&mut to_reorder);
+        } else {
+            let to_reorder = &mut state.windows;
+            let mut window_restored = vec![];
+            for window_handle in &state.window_order {
+                let pos = to_reorder
+                    .iter_mut()
+                    .position(|c| c.handle == *window_handle);
+                if let Some(pos) = pos {
+                    let c = to_reorder.remove(pos);
+                    window_restored.push(c);
```

Comment:
May I ask, why you chose `c` here? Maybe `i` like `index` might be more obvious?

Reply:
> I assume storing only the handlers was intended to be the cheaper operation, no?

Yes, I was going for the handlers because was the closest to a 'window id' I could find. You're saying that I could use `focus_manager` to get those handlers? 

I'm up to check the issue you related as a side note. I checked it a bit but so far I couldn't reproduce.
If you have a way to reproduce it, let me know!

Also I'm changing this PR to a draft because I just found out a bug while having 2 full screen window on different workspace. It seems that we should have a 'backup' for each workspace for the functionality to work correctly

Reply:
> Yes, I was going for the handlers because was the closest to a 'window id' I could find. You're saying that I could use `focus_manager` to get those handlers?

No, I meant, maybe to change `window_restored ` to `window_order_history` in a similar way as the focus manager does. Also using `VecDeque` had some benefits but I don't recall the specifics at the moment.

> I'm up to check the issue you related as a side note. I checked it a bit but so far I couldn't reproduce. If you have a way to reproduce it, let me know!

The focus restore issue occurs at most to me, when the `pinentry` window (having a window rule to make this window floating) opens to enter the passphrase for my signing key when I `git commit`.
 
> Also I'm changing this PR to a draft because I just found out a bug while having 2 full screen window on different workspace. It seems that we should have a 'backup' for each workspace for the functionality to work correctly

I guess storing the window order history would make most sense to keep for each tag, no?